File: HorseRacing.Contracts/Models/User/Responses/GetAccountBalance/GetAccountBalanceResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.User.Responses.GetAccountBalance
{
    public class GetAccountBalanceResponse : BaseResponse<GetAccountBalanceResponseDto>
    {
        public GetAccountBalanceResponse() : base() { }
        public GetAccountBalanceResponse(GetAccountBalanceResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/user/responses/get-account-balance/get-account-balance-response.ts
import type { BaseResponse } from '~/interfaces/api/contracts/base/responses/base-response';
import type { GetAccountBalanceResponseDto } from './get-account-balance-response-dto';

export class GetAccountBalanceResponse implements BaseResponse<GetAccountBalanceResponseDto> {
    Data: GetAccountBalanceResponseDto;

    constructor(data?: GetAccountBalanceResponseDto) {
        this.Data = data || { UserId: '', Balance: 0 };
    }
}
